### PR TITLE
modify the azure script with KF changes

### PR DESCRIPTION
**Pull request by suryakantkk**

### CODE REVIEW

**Comment by swapnil-kfuse** [comment]
@suryakantkk we need the ability to use the api_key when the authenticated ingestion is enabled similar to how it is there in original dd js file

File: scripts/azure_logs_integration_original_DD.js [hunk 1]
```diff
@@ -162,20 +159,20 @@ class Batcher {
...
 }
 
+// Provide Kloudfuse URL in quotes for "hostname" where you want to send the logs to
 class HTTPClient {
   constructor(context) {
     this.context = context;
     this.httpOptions = {
-      hostname: DD_HTTP_URL,
-      port: DD_HTTP_PORT,
+      hostname: "<Kloudfuse URL>",
```
Comment (pankaj-kf) [inline scripts/azure_logs_integration_original_DD.js:167]: Introduce a variable for this as well so its easier to set it.

File: scripts/azure_logs_integration_original_DD.js [hunk 2]
```diff
@@ -646,12 +669,6 @@ class EventhubLogHandler {
 }
 
 module.exports = async function (context, eventHubMessages) {
-  if (!DD_API_KEY || DD_API_KEY === "<DATADOG_API_KEY>") {
```
Comment (pankaj-kf) [inline scripts/azure_logs_integration_original_DD.js:649]: At some point folks will ask for key support; might as well preserve this.

**Review by pankaj-kf — APPROVED** [review]
lgtm; feel free to check-in after addressing comments.

File: scripts/azure_logs_integration_original_DD.js (hunk 2 above)
Comment (suryakantkk) [reply scripts/azure_logs_integration_original_DD.js:649]: do you know what would be the right header for the api key for us? Would it be `Kf_Api_Key`? (cc @vinayk-kfuse )

File: scripts/azure_logs_integration_original_DD.js (hunk 1 above)
Comment (suryakantkk) [reply scripts/azure_logs_integration_original_DD.js:167]: okay

File: scripts/azure_logs_integration_original_DD.js (hunk 2 above)
Comment (suryakantkk) [reply scripts/azure_logs_integration_original_DD.js:649]: After discussing with @harold-kfuse, added "KF_API_KEY_ENV" in the header for authentication.

File: scripts/azure/index.js [hunk 3]
```diff
@@ -166,16 +165,16 @@ class HTTPClient {
...
+      hostname: KF_URL,
+      port: KF_HTTP_PORT,
       path: "/api/v2/logs",
       method: "POST",
       headers: {
         "Content-Type": "application/json",
-        "DD-API-KEY": DD_API_KEY,
-        "DD-EVP-ORIGIN": "azure",
+        "KF-EVP-ORIGIN": "azure",
+        "KF_API_KEY_ENV": KF_API_KEY,
```
Comment (harold-kfuse) [inline scripts/azure/index.js:175]: Don't think this will work.
This comes in as DD payload, correct?
Ingester uses DD-API-KEY.

Comment (suryakantkk) [reply scripts/azure/index.js:175]: Yes. Okay.

File: scripts/azure_logs_integration_original_DD.js (hunk 2 above)
Comment (suryakantkk) [reply scripts/azure_logs_integration_original_DD.js:649]: changed it to DD_API_KEY

File: scripts/azure/index.js [hunk 4]
```diff
@@ -166,16 +165,16 @@ class HTTPClient {
...
+      hostname: KF_URL,
+      port: KF_HTTP_PORT,
       path: "/api/v2/logs",
       method: "POST",
       headers: {
         "Content-Type": "application/json",
-        "DD-API-KEY": DD_API_KEY,
-        "DD-EVP-ORIGIN": "azure",
+        "KF-EVP-ORIGIN": "azure",
+        "DD_API_KEY": KF_API_KEY,
```
Comment (harold-kfuse) [inline scripts/azure/index.js:175]: DD-API-KEY

**Comment by suryakantkk** [comment]
Tested successfully on ashish-dev namespace with auth enabled.

**Pull request merged**